pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -275,8 +277,8 @@ func doNetworkFilterListOperation(patchContext networking.EnvoyFilter_PatchConte
 				copy(fc.Filters[insertPosition+1:], fc.Filters[insertPosition:])
 				fc.Filters[insertPosition] = clonedVal
 			}
-		} else if cp.Operation == networking.EnvoyFilter_Patch_INSERT_BEFORE || cp.Operation == networking.EnvoyFilter_Patch_INSERT_FIRST {
-			// insert before/first without a filter match is same as insert in the beginning
+		} else if cp.Operation == networking.EnvoyFilter_Patch_INSERT_BEFORE {
+			// insert before without a filter match is same as insert in the beginning
```

Comment:
This looks not quite right. I know it was this behavior since the beginning. But deserves discussing.